core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -486,4 +492,17 @@ public InputStream getInputStream() throws IOException {
         }
         return null;
     }
+
+    private ProtectionDomain createProtectionDomainWithClasspath(String name) {
+        URL url = null;
+        try {
+            url = new URL("file:///" + getPackageNameFromClassName(name).replace('.', '/'));
```

Comment:
Will is cause problems to have these non-existant url's as the location? I would have thought using the first location from applicationClassDirectories would make more sense.

Reply:
Frameworks sometimes (right or wrong) use the code source to introspect the class path, so it should definitely be a real path whenever possible.

There should only be one protection domain (and thus one code source, and thus one URL) instance for each artifact.  If all of the artifacts are squashed into one place, then there should just be one protection domain instance for the whole class loader.

Also I would recommend using `java.net.URI` with a component constructor, calling `toURL()` on the result, to avoid encoding issues and that sort of thing.